apps/prairielearn/src/components/Navbar.html.ts

```diff
@@ -0,0 +1,853 @@
+          : '';
+
+  return html`
+    <ul
+      class="nav navbar-nav"
+      id="username-nav"
+      data-access-as-administrator="${access_as_administrator?.toString()}"
+      data-view-type="${viewType ?? null}"
+      data-authn-course-role="${authz_data?.authn_course_role}"
+      data-authn-course-instance-role="${authz_data?.authn_course_instance_role}"
+      data-has-instructor-access="${authz_data?.user_with_requested_uid_has_instructor_access_to_course_instance?.toString()}"
+    >
+      <li class="nav-item dropdown mb-2 mb-md-0 mr-2 ${navPage === 'effective' ? 'active' : ''}">
+        <a
+          class="nav-link dropdown-toggle"
+          href="#"
+          id="navbarDropdown"
+          role="button"
+          data-toggle="dropdown"
+          aria-haspopup="true"
+          aria-expanded="false"
+        >
+          ${displayedName} ${userBadge}
+          ${news_item_notification_count
+            ? html`
+                <span class="badge badge-pill badge-primary news-item-count">
+                  ${news_item_notification_count}
+                </span>
+              `
+            : ''}
+        </a>
+        <div class="dropdown-menu dropdown-menu-right" aria-labelledby="navbarDropdown">
+          ${authn_is_administrator
+            ? html`
+                <button type="button" class="dropdown-item" id="navbar-administrator-toggle">
+                  ${access_as_administrator
+                    ? 'Turn off administrator access'
+                    : 'Turn on administrator access'}
+                </button>
+
+                <div class="dropdown-divider"></div>
+              `
+            : ''}
+          ${ViewTypeMenu({ resLocals })} ${AuthnOverrides({ resLocals, navbarType })}
+          ${authz_data?.mode === 'Exam'
+            ? html`
+                <div class="dropdown-item-text">
+                  Exam mode means you have a checked-in reservation to CBTF. Your interactions with
+                  PrairieLearn are limited to your checked-in exam for the duration of your
+                  reservation.
+                </div>
+                <div class="dropdown-divider"></div>
+              `
+            : ''}
+          ${!authz_data || authz_data?.mode !== 'Exam'
+            ? html`
+                <a class="dropdown-item" href="${config.urlPrefix}/request_course">
+                  Course Requests
+                </a>
+              `
+            : ''}
+          <a class="dropdown-item" href="${config.urlPrefix}/settings">Settings</a>
+          <a
+            class="dropdown-item news-item-link"
+            href="${urlPrefix}/news_items"
+            ${news_item_notification_count
+              ? html`
+                  title="News (${news_item_notification_count} unread)" aria-label="News
+                  (${news_item_notification_count} unread)"
+                `
+              : html`title="News" aria-label="News"`}
```

Comment:
I'd probably structure this as `title=${news_item_notification_count ? 'News (...)' : 'News'` so that the formatting is a little nicer.

Aside, I've thought about introducing a little convenience API like this:

```ts
html`
  <a ${htmlAttributes(news_item_notification_count ? { ... } : { ... })}>...</a>
`
```

Basically as a bandaid over the fact that Prettier doesn't like to handle embedded key/value pairs in an `html` tagged template literal. This could strip out null/undefined values too.